patches/minecraft/net/minecraft/item/ItemSword.java.patch

```diff
@@ -1,13 +1,11 @@
 --- ../src-base/minecraft/net/minecraft/item/ItemSword.java
 +++ ../src-work/minecraft/net/minecraft/item/ItemSword.java
-@@ -88,7 +88,9 @@
+@@ -88,7 +88,7 @@
  
      public boolean func_82789_a(ItemStack p_82789_1_, ItemStack p_82789_2_)
      {
 -        return this.field_150933_b.func_150995_f() == p_82789_2_.func_77973_b() ? true : super.func_82789_a(p_82789_1_, p_82789_2_);
-+        ItemStack mat = this.field_150933_b.getRepairItemStack();
-+        if (!mat.func_190926_b() && net.minecraftforge.oredict.OreDictionary.itemMatches(mat, p_82789_2_, false)) return true;
-+        return super.func_82789_a(p_82789_1_, p_82789_2_);
++        return !this.field_150933_b.getRepairItemStack().func_190926_b() ? true : super.func_82789_a(p_82789_1_, p_82789_2_);
```

Comment:
You removed the `OreDictionary.itemMatches` check here, that's not right.
The check you have here will accept any material as long as a repair material exists.
It needs to compare the `p_82789_2_` parameter with the repair material using `OreDictionary.itemMatches`.